From 692910f67df33ec8e359c9149153b45c1b779648 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Thu, 19 Nov 2020 13:29:03 +0800
Subject: [PATCH] pfe_compiler: add GCC version 10.2.0 support

Add GCC version 10.2.0 support, to fix below compile error:

| pfe/0.9.0-r0/git/sw/pfe_platform/public/pfe_compiler.h:117:3:
| error: #error Please specify your compiler behavior by defining PFE_COMPILER_BITFIELD_BEHAVIOR.
|   117 |  #error Please specify your compiler behavior by defining PFE_COMPILER_BITFIELD_BEHAVIOR.
|       |   ^~~~~

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 sw/pfe_platform/public/pfe_compiler.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sw/pfe_platform/public/pfe_compiler.h b/sw/pfe_platform/public/pfe_compiler.h
index 895a27e..03609b0 100644
--- a/sw/pfe_platform/public/pfe_compiler.h
+++ b/sw/pfe_platform/public/pfe_compiler.h
@@ -90,6 +90,12 @@
 			/* Compiling driver */
 			#define PFE_COMPILER_BITFIELD_BEHAVIOR PFE_COMPILER_BITFIELD_HIGH_LAST
 		#endif
+	#elif ((__GNUC__ == 10) && (__GNUC_MINOR__ == 2) && (__GNUC_PATCHLEVEL__ == 0))
+                /* GCC version 10.2.0 */
+                #if (defined(PFE_CFG_TARGET_ARCH_aarch64))
+                        /* Compiling driver */
+                        #define PFE_COMPILER_BITFIELD_BEHAVIOR PFE_COMPILER_BITFIELD_HIGH_LAST
+                #endif
 	#endif
 #endif
 
-- 
2.17.1

